checker/check.ml

```diff
@@ -294,14 +294,22 @@ type intern_mode = Rec | Root | Dep (* Rec = standard, Root = -norec, Dep = depe
 (* Dependency graph *)
 let depgraph = ref LibraryMap.empty
 
-let marshal_in_segment f ch =
-  try
-    let stop = input_binary_int ch in
-    let v = Analyze.instantiate (Analyze.parse_channel ch) in
-    let digest = Digest.input ch in
+let marshal_in_segment ~validate ~value f ch =
```

Comment:
Maybe `value` could be passed as an option which means "validate with contents" if Some and "no validate" if None

Reply:
I thought about it, but it would make the code deciding to validate below a bit uglier.